sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcUtil.java

```diff
@@ -189,6 +190,18 @@ static JdbcIO.PreparedStatementSetCaller getPreparedStatementSetCaller(
           }
 
           String logicalTypeName = fieldType.getLogicalType().getIdentifier();
+
+          if (logicalTypeName.equals(MicrosInstant.IDENTIFIER)) {
+            // Process timestamp of MicrosInstant kind, which should only be passed from other type
+            // systems such as SQL and other Beam SDKs.
+            return (element, ps, i, fieldWithIndex) -> {
+              // MicrosInstant uses native java.time.Instant instead of joda.Instant.
+              java.time.Instant value =
+                  element.getLogicalTypeValue(fieldWithIndex.getIndex(), java.time.Instant.class);
+              ps.setTimestamp(i + 1, value == null ? null : new Timestamp(value.toEpochMilli()));
```

Comment:
I wish we had a more general solution for mapping these to java time vs. joda time

Reply:
Here is a mapping from java.time.Instant to java.sql.Timestamp, joda.time is not involved. The comment was a clarification because I see joda.time.Instant is usually used in the codebase (that's why I did not declare import Instant but write down the whole class name here)

Do you think we need a general Row.getJavaInstant() as  Row.getDateTime() ?

Reply:
Ah ok, my mistake. Regardless there's no action needed here, I was just lamenting that this date/time type situation is messy.